pinot-core/src/main/java/com/linkedin/pinot/core/segment/index/SegmentMetadataImpl.java

```diff
@@ -106,7 +107,16 @@ public SegmentMetadataImpl(File indexDir)
     _columnMetadataMap = new HashMap<String, ColumnMetadata>();
     _allColumns = new HashSet<String>();
     _schema = new Schema();
-    _indexDir = new File(indexDir, V1Constants.MetadataKeys.METADATA_FILE_NAME).getAbsoluteFile().getParent();
+
+    // For segment reload, we will load segment from the temporary index directory, but we need to set indexDir to the
+    // original index directory
+    String path = indexDir.getPath();
```

Comment:
rename indexDir as "indexDirToLoad", otherwise someone else doing a "cleanup"  might change line 121 to be _indexdir, which will work most of the time I suppose